### ISSUE TITLE
Setting new selected value post-mount

### ISSUE THREAD

**Issue by k5o** [AdeleD/react-paginate#62]
I'm attempting to manipulate the selected page number of PaginationBoxView without a user click (such as on pushState) after the component has been mounted. What I found was that the `selected` state wasn't updating because that state is only set before the initial mount.

I opened a PR with a simple solution using componentWillReceiveProps, let me know if this is appropriate: #61

**Comment by mortargrind** [comment]
Yes, i think this's a much needed functionality; really waiting for your this feature. But after having this, isn't it more convenient to change name of "initialSelected" to something else like "currentPage" or "activePage"? After all, that variable will not be used "only" on mounting phase. Or maybe rename both "initialSelected" and "forceSelected", just to make API more easy to understand?

**Comment by xiaolin** [comment]
I am also experiencing this issue. :+1:

**Comment by AdeleD** [comment]
After a "pushState" would you want the `clickCallback` to be called too? Or only change the `selected` state?

**Comment by xiaolin** [comment]
I think `clickCallback` should only fire on an actual `clickCallback`, if a "pushState" modifies the selected state, the user should handle the page update.

**Comment by AdeleD** [comment]
Ok so maybe you look for something like that :
```
componentWillReceiveProps: function(nextProps) {
  this.setState({
    selected: nextProps.selected
  });
}
```

I will test this as soon as I can.

**Comment by k5o** [comment]
@AdeleD yep that's what I was thinking. I have a PR up but there's a current merge conflict as it's a couple versions behind, I can work on it if you'd like: https://github.com/AdeleD/react-paginate/pull/61

**Comment by pabloflego** [comment]
Hi, What happened to this? Almost 2 years and was not addressed. Is this still being maintained?
Thanks!

**Comment by stfenjobs** [comment]
I am also experiencing this issue. 👍

**Comment by AdeleD** [comment]
You should be able to do what you want by updating the `forcePage` prop each time you want to change the selected page.
```javascript
<ReactPaginate forcePage={2} />
```

**Comment by AdeleD** [comment]
Hello,
In case it can help you, on version `6.5.0` of `react-paginate`, you can now define the event you want to listen onto.
You just need to instantiate the component with an `eventListener` prop.

For instance:
```javascript
<ReactPaginate eventListener="onMouseDown" />
```
